usr/src/lib/libzfs/common/libzfs_sendrecv.c

```diff
@@ -1662,26 +1702,32 @@ zfs_send_resume(libzfs_handle_t *hdl, sendflags_t *flags, int outfd,
 		 * If progress reporting is requested, spawn a new thread to
 		 * poll ZFS_IOC_SEND_PROGRESS at a regular interval.
 		 */
-		if (flags->progress) {
-			pa.pa_zhp = zhp;
-			pa.pa_fd = outfd;
-			pa.pa_parsable = flags->parsable;
-
-			error = pthread_create(&tid, NULL,
-			    send_progress_thread, &pa);
-			if (error != 0) {
-				zfs_close(zhp);
-				return (error);
-			}
+		pa.pa_zhp = zhp;
+		pa.pa_fd = outfd;
+		pa.pa_parsable = flags->parsable;
+		pa.pa_progress = flags->progress;
+		pa.pa_siginfo = flags->siginfo;
+
+		if (pa.pa_siginfo) {
+			/* Make sure this thread ignores signal, or ioctl aborts */
+			sigset_t sigset;
+			(void) sigemptyset(&sigset);
+			(void) sigaddset(&sigset, SIGINFO);
+			(void) pthread_sigmask(SIG_BLOCK, &sigset, NULL);
+		}
+
+		error = pthread_create(&tid, NULL,
+		    send_progress_thread, &pa);
```

Comment:
What if neither `progress` or `siginfo` is set?  Does the thread (correctly) do nothing then?

Reply:
Yes, it does create a thread that does nothing. It was simply a style decision to avoid having
```
if (progress || siginfo) phtread_create(...)
```
and same again for `phread_join()` at the end. But really I have no strong opinion either way.